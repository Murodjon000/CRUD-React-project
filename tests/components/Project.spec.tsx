import shallowSetup from "./mock.spec";
import toJson from "enzyme-to-json";

test("should render Project component", () => {
  const { projectWrapper, props } = shallowSetup();

  expect(projectWrapper.find("img").hasClass("test-project-img")).toBe(true);
  expect(projectWrapper.find(".test-project-title").text()).toBe(
    props.project[0].title
  );
  expect(projectWrapper.find(".test-project-desc").text()).toBe(
    props.project[0].description
  );
  expect(projectWrapper.find(".test-project-cat-btn").text()).toBe(
    props.project[0].category
  );
  expect(projectWrapper.find(".test-project-rm-btn").text()).toBe("Remove");
  expect(projectWrapper.find(".test-project-up-btn").text()).toBe("Update");
});

test("should render Project component Snapshot", () => {
  const { projectWrapper } = shallowSetup();

  expect(toJson(projectWrapper)).toMatchSnapshot();
});
