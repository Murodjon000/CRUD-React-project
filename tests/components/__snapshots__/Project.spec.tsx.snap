// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Details component Snapshot 1`] = `
<div
  className="project__details-wrapper"
>
  <div
    className="project__details-wrapper-image"
  >
    <img
      alt=""
      className="test-img"
      src="https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg"
    />
  </div>
  <div
    className="project__details-wrapper-box"
  >
    <h1
      className="test-title"
    >
      Where can I get some?
    </h1>
    <p
      className="test-desc"
    >
       
      ðŸ‘€ PageMaker including versions of Lorem Ipsum.
    </p>
    <button
      className="test-btn"
      onClick={[Function]}
      type="button"
    >
      Back to home
    </button>
  </div>
</div>
`;

exports[`should render Project component Snapshot 1`] = `
<div
  className="project__wrapper "
>
  <Link
    to="/project/1225232"
  >
    <div
      className="project__wrapper-image"
    >
      <img
        alt=""
        className="test-project-img"
        src="https://miro.medium.com/max/1000/1*9xsPgsjcFOnwhgPwtsO_HQ.jpeg"
      />
    </div>
  </Link>
  <div
    className="project__wrapper-info "
  >
    <h2
      className="test-project-title"
    >
      Where can I get some?
    </h2>
    <p
      className="text-link test-project-desc"
    >
      PageMaker including versions of Lorem Ipsum.
    </p>
    <div
      className="project__wrapper-info-bottom"
    >
      <button
        className="text-link test-project-cat-btn"
        type="button"
      >
        High
      </button>
      <span
        className="line-border"
      />
      <button
        className="text-link test-project-rm-btn"
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
      <span
        className="line-border"
      />
      <button
        className="text-link test-project-up-btn"
        onClick={[Function]}
        type="button"
      >
        Update
      </button>
    </div>
  </div>
</div>
`;
